### PR TITLE
fix(components): [date-picker] selected value comparison logic error

### LINKED ISSUE #15909: [Component] [date-picker] When type dates is given a default value, the panel expansion will not be selected the first time
Bug Type: **`Component`**

## Environment

- Vue Version: `3.4.19`
- Element Plus Version: `2.5.6`
- Browser / OS: `UserAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36`
- Build Tool: `Vite`

### Related Component

- `el-date-picker`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLWRhdGUtcGlja2VyXG4gICAgdi1tb2RlbD1cInZhbHVlNFwiXG4gICAgdHlwZT1cImRhdGVzXCJcbiAgICBwbGFjZWhvbGRlcj1cIlBpY2sgb25lIG9yIG1vcmUgZGF0ZXNcIlxuICAvPlxuPC90ZW1wbGF0ZT5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5jb25zdCB2YWx1ZTQgPSByZWYoW25ldyBEYXRlKCldKVxuPC9zY3JpcHQ+IiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiX28iOnt9fQ==)

### Steps to reproduce

When you open the panel, the default date is not selected, both dates will be selected when you click on a new date

录制_2024_02_20_19_29_38_547

### What is Expected?

Select the default value the first time you open it

### What is actually happening?

Not selected at first time

### Additional comments

https://github.com/element-plus/element-plus/blob/dev/packages/components/date-picker/src/composables/use-basic-date-table.ts#L137

should be sth like this? `d => d.isSame(cell.dayjs, 'month') && d.isSame(cell.dayjs, 'day') && d.isSame(cell.dayjs, 'year')`

### PR DESCRIPTION
**Pull request by betavs**
Please make sure these boxes are checked before submitting your PR, thank you!

fix #15909 

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Review by btea — APPROVED** [review]

**Comment by btea** [comment]
@betavs Could you add a test case?

**Comment by betavs** [comment]
> @betavs Could you add a test case?

done

File: packages/components/date-picker/src/composables/use-basic-date-table.ts [hunk 1]
```diff
@@ -133,9 +133,7 @@ export const useBasicDateTable = (
     const shouldIncrement = setDateText(cell, { count, rowIndex, columnIndex })
 
     const cellDate = cell.dayjs!.toDate()
-    cell.selected = _selectedDate.find(
-      (d) => d.valueOf() === cell.dayjs!.valueOf()
-    )
+    cell.selected = _selectedDate.find((d) => d.isSame(cell.dayjs, 'dates'))
```
Comment (kooriookami) [inline packages/components/date-picker/src/composables/use-basic-date-table.ts:136]: image

The second param of d.isSame seems has no `dates`. Did you check it?

Comment (kooriookami) [inline packages/components/date-picker/src/composables/use-basic-date-table.ts:136]:
```suggestion
    cell.selected = _selectedDate.find((d) => d.isSame(cell.dayjs, 'day'))
```

Comment (btea) [reply packages/components/date-picker/src/composables/use-basic-date-table.ts:136]: The code seems to support it.

https://github.com/iamkun/dayjs/blob/2076da970047c6e0a22c8d4243a42d52833a5df2/types/index.d.ts#L26

Comment (kooriookami) [reply packages/components/date-picker/src/composables/use-basic-date-table.ts:136]: I found it supports plurality. What the difference between `date` and `day`?

Comment (betavs) [reply packages/components/date-picker/src/composables/use-basic-date-table.ts:136]: Wouldn't it be better to use `date`?

Comment (kooriookami) [reply packages/components/date-picker/src/composables/use-basic-date-table.ts:136]: I asked the author of dayjs, `date`, `dates`, `day`, `days` are the same. In order to be compatible with momentjs. So, choose a semantically explicit value is better.

**Review by kooriookami — APPROVED** [review]

**Pull request merged**